pkg/providers/vsphere/vsphere.go

```diff
@@ -766,7 +767,7 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, datacenterSpec v1alpha1.VSphe
 func buildTemplateMapMD(clusterSpec *cluster.Spec, datacenterSpec v1alpha1.VSphereDatacenterConfigSpec, workerNodeGroupMachineSpec v1alpha1.VSphereMachineConfigSpec) map[string]interface{} {
 	bundle := clusterSpec.VersionsBundle
 	format := "cloud-config"
-	kubeletExtraArgs := clusterapi.SecureTlsCipherSuitesExtraArgs()
+	kubeletExtraArgs := clusterapi.SecureTlsCipherSuitesExtraArgs().Append(clusterapi.ResolvConfExtraArgs(clusterSpec.Spec.ClusterNetwork.DNS.ResolvConf))
```

Comment:
I'm wondering if we should add tests for these and the cp (also for docker)
Maybe we can just add the resolv config to an existing test that has `sharedExtraArgs`

Reply:
These methods are called through the GenerateCAPISpec methods for both create and upgrade (which we have tests for), but there aren't any tests to test this directly for CP and workers. So it is tested indirectly, but we can create an issue to add tests for this in a separate PR for additional testing

Reply:
Right, I meant if we should add cases to those tests with resolvConf configurations (or just add the resolvConf to some to some those tests)
If not, this code is run but we don't actually verify the validity of the output for a resolvConf case. Because we don't have a case with such input